### PR TITLE
[DO-1974] Fix saving nacp id of related persons

### PR DESCRIPTION
**Pull request by EmmaSamoilova**

### CODE REVIEW

**Review by Iva-khar — CHANGES_REQUESTED** [review]

**Review by Iva-khar — APPROVED** [review]

File: business_register/converter/declaration.py [hunk 1]
```diff
@@ -1021,45 +1030,36 @@ def find_city(self, address_data):
...
     # TODO: maybe we should simplify spouse to CharField with full name
-    def save_spouse(self, relatives_data, pep, declaration):
-        SPOUSE_TYPES = ['дружина', 'чоловік']
-        # TODO: decide should we store new Pep that not spouse from relatives_data
+    def save_related_person(self, relatives_data, pep, declaration):
         for relative_data in relatives_data:
-            if type(relative_data) != dict:
-                self.log_error(f'Invalid value: relative_data = {relative_data}')
-                continue
+            SPOUSE_TYPES = ['дружина', 'чоловік']
```
[the remark this answers is not in the document]
Comment (EmmaSamoilova) [reply business_register/converter/declaration.py]: done

File: business_register/converter/declaration.py [hunk 2]
```diff
@@ -1021,45 +1030,36 @@ def find_city(self, address_data):
...
+            for link in related_person_links:
+                related_person = link.to_person
+                related_person_nacp_id = relative_data.get('id')
+                try:
+                    same_name = is_same_full_name(relative_data, related_person)
+                except InvalidRelativeData as e:
+                    self.log_error(f'{e}')
+                    continue
+                else:
+                    if same_name:
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:1051]: this line can throw error: `same_name` not declared

File: business_register/migrations/0120_auto_20210615_1141.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                    SET nacp_id1 = ARRAY[nacp_id]
+                    WHERE nacp_id IS NOT NULL;
+                ''',
+            reverse_sql=f'''
+                    UPDATE business_register_pep
+                    SET nacp_id =
+                      (SELECT nacp_id1[1]
+                       FROM business_register_pep AS pep2
+                       WHERE business_register_pep.id=pep2.id)
+                    WHERE business_register_pep.nacp_id1 IS NOT NULL;
```
Comment (roman-tiukh) [inline business_register/migrations/0120_auto_20210615_1141.py:40]: maybe?
```suggestion
                    UPDATE business_register_pep
                    SET nacp_id = nacp_id1[1]
                    WHERE business_register_pep.nacp_id1 IS NOT NULL;
```

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: business_register/migrations/0120_auto_20210615_1141.py (hunk 3 above)
Comment (EmmaSamoilova) [reply business_register/migrations/0120_auto_20210615_1141.py]: done

**Review by roman-tiukh — APPROVED** [review]

**Review by OlexandrTopuzov — APPROVED** [review]

**Pull request merged**